WalletWasabi.Fluent/ZXing/SKBitmapLuminanceSource.cs

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2017 ZXing.Net authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Runtime.CompilerServices;
+using SkiaSharp;
+
+namespace ZXing.SkiaSharp
+{
+    /// <summary>
+    /// A luminance source class which consumes a Mat image from SkiaSharp and calculates the luminance values based on the bytes of the image
+    /// </summary>
+    public class SKBitmapLuminanceSource : BaseLuminanceSource
+    {
+        /// <summary>
+        /// initializing constructor
+        /// </summary>
+        /// <param name="image"></param>
+        public SKBitmapLuminanceSource(SKBitmap image)
+           : base(image.Width, image.Height)
```

Comment:
If it's a non ARM device then it's possible to have Windows on it. Taking your reaction into account, I guess, the answer is "no" (:

Let me then check all potential failure points and come back with questions or a special build.

One more thing here is that our custom code had one bitmap fetch before the actual start just because it's null. Maybe FlashCam doesn't handle that case.